src/components/ProfitAndLossDetailReport/ProfitAndLossDetailReport.tsx

```diff
@@ -0,0 +1,247 @@
+import { useContext, useState, useMemo, useCallback } from 'react'
+import { ProfitAndLoss } from '../ProfitAndLoss'
+import { useProfitAndLossDetailLines } from '../../hooks/useProfitAndLoss/useProfitAndLossDetailLines'
+import { useLayerContext } from '../../contexts/LayerContext'
+import { SourceDetailView } from '../LedgerAccountEntryDetails/LedgerAccountEntryDetails'
+import { BaseDetailView } from '../BaseDetailView/BaseDetailView'
+import { DataTable, type ColumnConfig } from '../DataTable/DataTable'
+import { Badge } from '../Badge'
+import { DateTime } from '../DateTime'
+import { TextSize, TextWeight } from '../Typography'
+import { DetailsList, DetailsListItem } from '../DetailsList'
+import { DataState, DataStateStatus } from '../DataState/DataState'
+import { Button } from '../ui/Button/Button'
+import { VStack } from '../ui/Stack/Stack'
+import { format } from 'date-fns'
+import type { LedgerEntrySource } from '../../types/ledger_accounts'
+import { Direction } from '../../types'
+import { BreadcrumbItem, DetailReportBreadcrumb } from '../DetailReportBreadcrumb/DetailReportBreadcrumb'
+import type { PnlDetailLine, LedgerEntrySourceType } from '../../hooks/useProfitAndLoss/useProfitAndLossDetailLines'
+import { MoneySpan } from '../ui/Typography/MoneyText'
+
+const COMPONENT_NAME = 'ProfitAndLossDetailReport'
+
+/* Our source detail component expects an old schema.
+ * This converts for backwards compatibility until we switch that component to our new schemas with fixed variable types. */
+const convertSourceForDetailView = (source: LedgerEntrySourceType): LedgerEntrySource => {
+  return {
+    display_description: source.displayDescription,
+    entity_name: source.entityName,
+    type: source.type,
+  }
+}
+
+enum PnlDetailColumns {
+  Date = 'Date',
+  Type = 'Type',
+  Account = 'Account',
+  Description = 'Description',
+  Amount = 'Amount',
+  Balance = 'Balance',
+}
+
+export interface ProfitAndLossDetailReportStringOverrides {
+  title?: string
+  dateColumnHeader?: string
+  typeColumnHeader?: string
+  accountColumnHeader?: string
+  descriptionColumnHeader?: string
+  amountColumnHeader?: string
+  balanceColumnHeader?: string
+  sourceDetailsTitle?: string
+}
+
+export interface ProfitAndLossDetailReportProps {
+  lineItemName: string
+  breadcrumbPath?: BreadcrumbItem[]
+  onClose: () => void
+  onBreadcrumbClick?: (lineItemName: string) => void
+  stringOverrides?: ProfitAndLossDetailReportStringOverrides
+}
+
+const ErrorState = () => (
+  <DataState
+    status={DataStateStatus.failed}
+    title='Error loading detail lines'
+    description='There was an error loading the profit and loss detail lines'
+  />
+)
+
+const EmptyState = () => (
+  <DataState
+    spacing={true}
+    status={DataStateStatus.info}
+    title='No detail lines found'
+    description='There are no detail lines for this profit and loss item'
+  />
+)
```

Comment:
I think we'll want spacing on both!
```suggestion
const ErrorState = () => (
  <DataState
    spacing
    status={DataStateStatus.failed}
    title='Error loading detail lines'
    description='There was an error loading the profit and loss detail lines'
  />
)

const EmptyState = () => (
  <DataState
    spacing
    status={DataStateStatus.info}
    title='No detail lines found'
    description='There are no detail lines for this profit and loss item'
  />
)
```